language-support/ts/daml-ledger/index.test.ts

```diff
@@ -0,0 +1,244 @@
+  return {
+    templateId: "foo-id",
+    contractId: coid.toString() as ContractId<object>,
+    signatories: [],
+    observers: [],
+    agreementText: "fooAgreement",
+    key: undefined,
+    payload: {},
+  };
+};
+
+const fooEvent = (coid: number): Event<Foo, undefined, "foo-id"> => {
+  return { created: fooCreateEvent(coid) };
+};
+
+
+const fooArchiveEvent = (coid: number): Event<Foo, undefined, "foo-id"> => {
+  return {
+    archived: {
+      templateId: "foo-id",
+      contractId: coid.toString() as ContractId<object>,
+    },
+  };
+};
+
+const mockOptions = {
+  token: "dummyToken",
+  httpBaseUrl: "http://localhost:5000/",
+  wsBaseUrl: "ws://localhost:4000/",
+};
+
+afterEach(() => {
+  WS.clean();
+});
+
+describe("streamQuery", () => {
+  // we only check that nothing crashes.
+  test("receive unknown message", async () => {
+    const server = new WS("ws://localhost:4000/v1/stream/query", {
+      jsonProtocol: true,
+    });
+    const ledger = new Ledger(mockOptions);
+    const stream = ledger.streamQuery(Foo);
+    await server.connected;
+    console.log("connected");
+    stream.on("change", (evs) => console.log(evs));
+    stream.on("close", (ev) => console.log(ev));
+    stream.off("change", (evs) => console.log(evs));
+    stream.off("close", (ev) => console.log(ev));
+    server.send("mickey mouse");
+  });
+
+  // we only check that nothing crashes.
+  test("receive warnings", async () => {
+    const server = new WS("ws://localhost:4000/v1/stream/query", {
+      jsonProtocol: true,
+    });
+    const ledger = new Ledger(mockOptions);
+    const stream = ledger.streamQuery(Foo);
+    await server.connected;
+    console.log("connected");
+    stream.on("change", (evs) => console.log(evs));
+    stream.on("close", (ev) => console.log(ev));
+    stream.off("change", (evs) => console.log(evs));
+    stream.off("close", (ev) => console.log(ev));
+    server.send({ warnings: ["oh oh"] });
+  });
+
+  // we only check that nothing crashes.
+  test("receive errors", async () => {
+    const server = new WS("ws://localhost:4000/v1/stream/query", {
+      jsonProtocol: true,
+    });
+    const ledger = new Ledger(mockOptions);
+    const stream = ledger.streamQuery(Foo);
+    await server.connected;
+    console.log("connected");
+    stream.on("change", (evs) => console.log(evs));
+    stream.on("close", (ev) => console.log(ev));
+    stream.off("change", (evs) => console.log(evs));
+    stream.off("close", (ev) => console.log(ev));
+    server.send({ errors: ["not good!"] });
+  });
+
+  test("receive empty events", async () => {
+    let receivedEvents: object[] = [];
+    const server = new WS("ws://localhost:4000/v1/stream/query", {
+      jsonProtocol: true,
+    });
+    const ledger = new Ledger(mockOptions);
+    const stream = ledger.streamQuery(Foo);
+    await server.connected;
+    console.log("connected");
+    stream.on("change", (evs) => (receivedEvents = [...evs]));
```

Comment:
Wouldn't it make sense to use `jest.fn` here and check that it gets called the way expect it to be called?

Reply:
both work, I changed it to the a mock function.